### PR TITLE
Clone repositories in parallel

### PR DESCRIPTION
**Pull request by kanurag94**
Addresses #826

### CODE REVIEW

File: scripts/microannotate_generator.py [hunk 1]
```diff
@@ -25,7 +26,8 @@ def __init__(self, cache_root, repo_url, tokenize, remove_comments):
         self.repo_dir = os.path.join(cache_root, "mozilla-central")
 
     def generate(self):
-        repository.clone(self.repo_dir)
+        executor = concurrent.futures.ThreadPoolExecutor(max_workers=os.cpu_count() + 1)
```
Comment (marco-c) [inline scripts/microannotate_generator.py]: We're only executing two tasks, what is the point of having so many workers? :P

Comment (marco-c) [inline scripts/microannotate_generator.py]: You shoud use the executor as a context manager to make sure the threads are cleaned up at the end.

File: scripts/microannotate_generator.py [hunk 2]
```diff
@@ -37,9 +39,11 @@ def generate(self):
         )
         git_repo_path = os.path.basename(self.repo_url)
 
-        retry(
-            lambda: subprocess.run(
-                ["git", "clone", self.repo_url, git_repo_path], check=True
+        executor.submit(
```
Comment (marco-c) [inline scripts/microannotate_generator.py]: You are passing to `executor.submit` the result of executing the clone, this is not going to work.

**Comment by marco-c** [comment]
Re-request review when you're done with the changes.

Note: the README update should be done in a separate PR.

**Comment by kanurag94** [comment]
@marco-c This is working locally. Can you please re review the PR again. I will remove the readme from this one

File: scripts/microannotate_generator.py [hunk 3]
```diff
@@ -29,37 +30,20 @@ def __init__(self, cache_root, repo_url, tokenize, remove_comments):
         self.repo_dir = os.path.join(cache_root, "mozilla-central")
 
     def generate(self):
-        repository.clone(self.repo_dir)
 
-        logger.info("mozilla-central cloned")
+        with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:
+            executor.submit(repository.clone, self.repo_dir)
+            logger.info("Cloning mozilla-central")
```
Comment (marco-c) [inline scripts/microannotate_generator.py]: You can keep the "mozilla-central cloned" log at the end of cloning, by using https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.add_done_callback on the future returned by executor.submit

Comment (kanurag94) [reply scripts/microannotate_generator.py]: Thanks, I have updated PR.
Can you please re review ?

**Pull request merged**